customagg/functions.py

```diff
@@ -0,0 +1,46 @@
+import re
+import logging
+import datetime as dt
+import math
+import numpy as np
+import pandas as pd
+
+from iotfunctions.base import BaseSimpleAggregator
+from iotfunctions.ui import (UIMultiItem,UIExpression)
+
+logger = logging.getLogger(__name__)
+
+# Specify the URL to your package here.
+# This URL must be accessible via pip install
+
+PACKAGE_URL = 'git+https://github.com/ankit-jha/addCustomIotFn@starter_agg_package'
+
+class HelloWorldAggregator(BaseSimpleAggregator):
+    '''
+    The docstring of the function will show as the function description in the UI.
+    '''
+
+    def __init__(self, source=None, expression=None):
+        if expression is None or not isinstance(expression, str):
+            raise RuntimeError("argument expression must be provided and must be a string")
+
+        self.input_items = source
+        self.expression = expression
+
+    def execute(self, group):
+        return eval(re.sub(r"\$\{GROUP\}", r"group", self.expression))
+
+    @classmethod
+    def build_ui(cls):
+        inputs = []
+        # Input variable name must be kept 'source'
+        # Output variable name must be kept 'name'
+        inputs.append(UIMultiItem(name='source', datatype=None, description=('Choose the data items'
+                                                                            ' that you would like to'
+                                                                                  ' aggregate'),
+                                  output_item='name', is_output_datatype_derived=True))
+
+        inputs.append(UIExpression(name='expression', description='Use ${GROUP} to reference the current grain.'
+                                                    'All Pandas Series methods can be used on the grain.'
+                                                    'For example, ${GROUP}.max() - ${GROUP}.min().'))
+        return (inputs, [])
```

Comment:
I can not see `output` parameter here. Where the data will be stored?  Have you run this function in the pipeline ?

Reply:
Output metadata is created in the below line and data gets stored the same way as it is for other functions
```python
        output_item='name', is_output_datatype_derived=True))
```
This is currently running at https://dashboard-dev.connectedproducts.internetofthings.ibmcloud.com/explore/872_Sample_Boiler_Type/data/Daily_max_input_flow_rate

The function is being picked up from sample.py
https://github.com/ibm-watson-iot/functions/blob/development/iotfunctions/sample.py#L836

Reply:
![image](https://user-images.githubusercontent.com/18572477/81647714-1ab8d400-944b-11ea-9904-4275ef0f76b5.png)


If you not provide output then there will be something which is not meaning full. In the output field its saying to select data item.

Reply:
Since it is custom function please do not add it to  sample.py file.  (https://github.com/ibm-watson-iot/functions/blob/development/iotfunctions/sample.py#L836) We don't need that in iotfunction code.

Reply:
I added that at @sedgewickmm18 suggestion. 
I can remove it but ain't sample.py is for sample functions for later reference